scripts/irods/test/test_ibun.py

```diff
@@ -100,3 +101,80 @@ def test_ibun_extraction_of_big_tar_file__issue_4118(self):
+                lib.make_file(local_file, filesize, 'random')
+                out,_ = lib.execute_command(['ls', '-l', local_file])
+                print(out)
+
+                # generate the collection to bundle up
+                self.admin.assert_icommand(['imkdir', collection_path])
+                self.admin.assert_icommand(['iput', '-f', '-R', replication, local_file, put_path])
+
+                # bundle the collection to an archive file in the replication hierarchy
+                self.admin.assert_icommand(['ibun', '-c', '-Dbzip2', '-R', replication, archive_path, collection_path])
+                print(self.admin.run_icommand(['ils', '-l', archive_path])[0])
+
+                # ensure that the replicas were created and are good
+                query = 'select DATA_SIZE, DATA_REPL_STATUS, DATA_REPL_NUM where COLL_NAME = \'{}\' and DATA_NAME = \'{}\''.format(
+                    os.path.dirname(archive_path), os.path.basename(archive_path))
+
+                out, err, ec = self.admin.run_icommand(['iquest', '%s...%s...%s', query])
+
+                self.assertEqual(0, ec)
+                self.assertEqual(0, len(err))
+                self.assertEqual(2, len(out.splitlines()))
+
+                for result in out.splitlines():
+                    r = result.split('...')
+                    self.assertEqual(compressed_size, int(r[0]))
+                    self.assertEqual(1, int(r[1]))
+
+                # overwrite archive file via a forced ibun of the same collection to the same logical path
+                self.admin.assert_icommand(['ibun', '-c', '-Dbzip2', '-f', '-R', replication, archive_path, collection_path])
+                print(self.admin.run_icommand(['ils', '-l', archive_path])[0])
+
+                # ensure that the replicas were overwritten and are good
+                out, err, ec = self.admin.run_icommand(['iquest', '%s...%s...%s', query])
+
+                self.assertEqual(0, ec)
+                self.assertEqual(0, len(err))
+                self.assertEqual(2, len(out.splitlines()))
+
+                for result in out.splitlines():
+                    r = result.split('...')
+                    self.assertEqual(compressed_size, int(r[0]))
+                    self.assertEqual(1, int(r[1]))
+
+            finally:
+                self.admin.run_icommand(['irm', '-r', '-f', collection_path])
+                self.admin.run_icommand(['irm', '-f', archive_path])
+                lib.remove_child_resource(replication, resource_0, self.admin)
+                lib.remove_child_resource(replication, resource_1, self.admin)
+                lib.remove_resource(replication, self.admin)
+                lib.remove_resource(resource_0, self.admin)
+                lib.remove_resource(resource_1, self.admin)
+
+        do_test_ibun_atop_existing_archive_file_in_replication_hierarchy(self, 1039425, 1044480)
+        do_test_ibun_atop_existing_archive_file_in_replication_hierarchy(self, 1039426, 1054720)
```

Comment:
*sometimes* this one byte size delta is not enough... so might want to make it 5 bytes or so between 'under' and 'over' scenarios.

Reply:
Okay. Should I bump back the smaller size as well to ensure that the archive file stays under 1 MiB? I'm not quite sure what the threshold is. Does it depend on the libarchive version and/or underlying machinery in the OS?

From https://github.com/irods/irods/issues/5426#issuecomment-782907963:

> The bug is triggered as soon as the archive file is larger than 1 mebibyte (1024*1024 = 1048576 bytes).

Reply:
I bumped by 1KB each in either direction and test passes for me